elastic4s-domain/src/main/scala/com/sksamuel/elastic4s/requests/snapshots/response.scala

```diff
@@ -1,11 +1,22 @@
 package com.sksamuel.elastic4s.requests.snapshots
 
-import com.fasterxml.jackson.annotation.JsonProperty
-
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonSubTypes, JsonTypeInfo}
+import com.sksamuel.elastic4s.requests.common.Shards
 import scala.concurrent.duration._
 
 case class CreateRepositoryResponse(acknowledged: Boolean)
-case class CreateSnapshotResponse(accepted: Boolean)
+@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
+@JsonSubTypes(
+  Array(
+    new JsonSubTypes.Type(value = classOf[CreateSnapshotResponseAsync], name = "accepted"),
+    new JsonSubTypes.Type(value = classOf[CreateSnapshotResponseAwait], name = "snapshot")))
+sealed trait CreateSnapshotResponse { def succeeded: Boolean }
+case class CreateSnapshotResponseAsync(accepted: Boolean) extends CreateSnapshotResponse {
```

Comment:
Naming could probably be improved here 😅 I'm open to suggestions

Reply:
I think these names are fine, they're clear.